.github/workflows/current-workflow.yml

```diff
@@ -31,25 +43,82 @@ jobs:
-        run: docker inspect flask_backend_container
+
+      - name: Inspect ${{ env.BACKEND_IMAGE_NAME }}
+        run: docker inspect ${{ env.BACKEND_IMAGE_NAME }}-${{ env.BRANCH_NAME }}
+
       - name: Stop container
-        run: docker stop flask_backend_container
+        run: docker stop ${{ env.BACKEND_IMAGE_NAME }}-${{ env.BRANCH_NAME }}
+
       - name: Restart container
-        run: docker restart flask_backend_container
+        run: docker restart ${{ env.BACKEND_IMAGE_NAME }}-${{ env.BRANCH_NAME }}
+
       - name: Show container status
         run: docker ps -a
+
       - name: Stop container again
-        run: docker stop flask_backend_container
+        run: docker stop ${{ env.BACKEND_IMAGE_NAME }}-${{ env.BRANCH_NAME }}
+
       - name: Remove container
-        run: docker rm flask_backend_container
+        run: docker rm ${{ env.BACKEND_IMAGE_NAME }}-${{ env.BRANCH_NAME }}
         
       - run: echo "Docker-and-Python-Flask-Service job's status is ${{ job.status }}."
+
+  Publish-Flask-Image-to-GitHub-Package-Registry:
+    name: Publish Docker Python Flask Image to GitHub Package Registry
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+      attestations: write
+      id-token: write
+    steps:
+      - run: echo "The job was automatically triggered by a ${{ github.event_name }} event."
+      - run: echo "Job is running on a ${{ runner.os }} server hosted by GitHub"
+      - run: echo "Branch name is ${{ github.ref }} and repository is ${{ github.repository }}."
+
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - run: echo "The repository ${{ github.repository }} has been cloned to runner."
+      - run: echo "Workflow is now ready to test code on runner."
+
+      - name: Sanitize branch name
+        run: echo "BRANCH_NAME=$(echo '${{ github.ref_name }}' | sed 's/[^a-zA-Z0-9._-]/-/g')" >> $GITHUB_ENV
+
+      - name: Install required dependencies
+        run: |
+          cd ./code/backend
+          pip install -r requirements.txt
+      
+      - name: Login to GitHub Packages registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract tag data for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}/${{ env.BACKEND_IMAGE_NAME }}:${{ env.BRANCH_NAME }}
+
+      - name: Build and push the Docker Image
+        id: push
+        uses: docker/build-push-action@v5
+        with:
+          context: ./code/backend
+          push: true
+          tags: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}/${{ env.BACKEND_IMAGE_NAME }}:${{ env.BRANCH_NAME }}
+
+      - run: echo "Publish-Flask-Image-to-GitHub-Package-Registry job's status is ${{ job.status }}."
   
   Node-Service:
```

Comment:
This stage is no longer needed and you may remove it .
There should be only four stages when this compelte